x/protocolpool/keeper/msg_server_test.go

```diff
@@ -0,0 +1,223 @@
+	}{
+		"empty recipient address": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        suite.poolKeeper.GetAuthority(),
+				RecipientAddress: "",
+				TotalBudget:      &fooCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         2,
+				Period:           60,
+			},
+			expErr:    true,
+			expErrMsg: "empty address string is not allowed",
+		},
+		"empty authority": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        "",
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &fooCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         2,
+				Period:           60,
+			},
+			expErr:    true,
+			expErrMsg: "empty address string is not allowed",
+		},
+		"invalid authority": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        "invalid_authority",
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &fooCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         2,
+				Period:           60,
+			},
+			expErr:    true,
+			expErrMsg: "invalid authority",
+		},
+		"invalid budget": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        suite.poolKeeper.GetAuthority(),
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &invalidCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         2,
+				Period:           60,
+			},
+			expErr:    true,
+			expErrMsg: "total budget cannot be zero",
+		},
+		"invalid start time": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        suite.poolKeeper.GetAuthority(),
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &fooCoin,
+				StartTime:        -10,
+				Tranches:         2,
+				Period:           60,
+			},
+			expErr:    true,
+			expErrMsg: "invalid start time",
+		},
+		"invalid tranches": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        suite.poolKeeper.GetAuthority(),
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &fooCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         0,
+				Period:           60,
+			},
+			expErr:    true,
+			expErrMsg: "tranches must be a positive value",
+		},
+		"invalid period": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        suite.poolKeeper.GetAuthority(),
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &fooCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         2,
+				Period:           0,
+			},
+			expErr:    true,
+			expErrMsg: "period length should be a positive value",
+		},
+		"all good": {
+			input: &types.MsgSubmitBudgetProposal{
+				Authority:        suite.poolKeeper.GetAuthority(),
+				RecipientAddress: recipientAddr.String(),
+				TotalBudget:      &fooCoin,
+				StartTime:        suite.ctx.BlockTime().Unix(),
+				Tranches:         2,
+				Period:           60,
+			},
+			expErr: false,
```

Comment:
Should a start time is the past be allowed? Retroactive funding could just do a community spend proposal for the whole chunk. Currently it is.

cc @likhita-809 @hxrts